compiler/rustc_data_structures/src/transitive_relation.rs

```diff
@@ -354,6 +354,20 @@ impl<T: Eq + Hash + Copy> TransitiveRelation<T> {
             .collect()
     }
 
+    /// Given an element A, elements B with the lowest index such that `A R B`
+    /// and `B R A`, or `A` if no such element exists.
+    pub fn scc_representative(&self, a: T) -> T {
+        match self.index(a) {
+            Some(a_i) => self.with_closure(|closure| {
+                closure
+                    .iter(a_i.0)
+                    .find(|i| closure.contains(*i, a_i.0))
+                    .map_or(a, |i| self.elements[i])
```

Comment:
Why would we not find such an element? In other words, what's the justification for the `map_or`, over an ` unwrap`? Shouldn't all free regions be represented in `TransitiveRelations`?

Reply:
Borrowck will always have this as a reflexive relation, but the data structure doesn't require that (maybe it should though).